documentation/smg/16-serialiser.md

```diff
@@ -0,0 +1,275 @@
+
+#### Object Array
+Object array data (tag 23) are serialised as:
+
+RANK |     tag    | dim1 | dim2 | ... |Data|     ...            |    ...     |   ...   |  ...
+-----|------------|------|------|:---:|----|--------------------|------------|---------|------
+NULL | 32 + tag   | 0    |      |     | 33 | length(class_name) | class_name |         |
+0    |    tag     |      |      |     | 33 | length(class_name) | class_name | ser_tag | Data |
+1    | 32 + tag   | nElem|      |     | 33 | length(class_name) | class_name | ser_tag | Data |
+2    | 64 + tag   | dim1 | dim2 |     | 33 | length(class_name) | class_name | ser_tag | Data |
+N    | N<<5 + tag | dim1 | dim2 | ... | 33 | length(class_name) | class_name | ser_tag | Data |
+
+Where
+`33` is a 1-D char tag, 
+`length(class_name)` is stored as a `uint32`, 
+`class_name` is a char string, 
+`ser_tag` is a `uint8` tag (described below) and
+`data` is the serialisation of the object by means described by `ser_tag`
+
+##### ser_tag
+The object array's `ser_tag` contains information about the means by which the object has been serialised.
+It can have one of three values:
+
+0) The object has been serialised by its own `serialize` method
+1) The object has been serialised by the `saveobj` method or function
+2) The object has been serialised by calling `struct` on the object
+
+#### Function handles
+Function handles are serialised differently depending on their type. 
+As function handles in MATLAB are purely scalar objects, the rank-component of the tag is utilised for tagging the type of function handle (thus the method to [de]serialise), these types are:
+
+Handle type         |   tag   | value
+--------------------|---------|-------
+Simple/Class simple | 32 + 25 | 57
+Anonymous           | 64 + 25 | 89
+Scoped              | 96 + 25 | 121
+
+Function handles are serialised as:
+Handle type         |   tag   | data                                       | ... 
+--------------------|---------|--------------------------------------------|-----------------------------------------
+Simple/Class simple |  57     | function name as serialised string         |
+Anonymous           |  89     | anonymous function as serialised string    | relevant workspace as serialised struct
+Scoped              |  121    | function parentage as serialised cell array| 
+
+### Tags
+Tag| Meaning
+---|----------------------
+  0| LOGICAL
+  1| CHAR
+  2| MATLAB_STRING
+  3| DOUBLE
+  4| SINGLE
+  5| INT8
+  6| UINT8
+  7| INT16
+  8| UINT16
+  9| INT32
+ 10| UINT32
+ 11| INT64
+ 12| UINT64
+ 13| COMPLEX_DOUBLE
+ 14| COMPLEX_SINGLE
+ 15| COMPLEX_INT8
+ 16| COMPLEX_UINT8
+ 17| COMPLEX_INT16
+ 18| COMPLEX_UINT16
+ 19| COMPLEX_INT32
+ 20| COMPLEX_UINT32
+ 21| COMPLEX_INT64
+ 22| COMPLEX_UINT64
+ 23| CELL
+ 24| STRUCT
+ 25| FUNCTION_HANDLE
+ 26| VALUE_OBJECT
+ 27| HANDLE_OBJECT_REF
+ 28| ENUM
+ 29| SPARSE_LOGICAL
+ 30| SPARSE_DOUBLE
+ 31| SPARSE_COMPLEX_DOUBLE
+
+### Limitations
```

Comment:
The anonymous function limitation in the old format appears to have disappeared - may be worth commenting on (and any other changes I missed.)